cmake/buildinfo/buildinfo.cmake

```diff
@@ -0,0 +1,29 @@
+cmake_host_system_information(RESULT CPU_NAME QUERY PROCESSOR_NAME)
+cmake_host_system_information(RESULT CPU_DESC QUERY PROCESSOR_DESCRIPTION)
+
+add_custom_command(
+	OUTPUT cmake_info.json
+	COMMAND ${CMAKE_COMMAND} -E capabilities >> cmake_info.json
+	COMMENT "Generating ${NBL_ROOT_PATH_BINARY}/cmake_info.json"
+)
+
+set(VKSDK_INFO_COMMAND "$ENV{VULKAN_SDK}/bin/vulkaninfoSDK")
```

Comment:
don't assume the ENV [exists](<https://cmake.org/cmake/help/v3.31/module/FindVulkan.html>)

```cmake
cmake_path(GET Vulkan_INCLUDE_DIR PARENT_PATH VULKAN_SDK_INSTALL_DIRECTORY)
set(VKSDK_INFO_COMMAND "${VULKAN_SDK_INSTALL_DIRECTORY}/bin/vulkaninfoSDK")
```

you should also check if vulkan driver is enabled at all _first_ (`if (Vulkan_FOUND)`), if not then pack empty "vulkan-driver-disabled.txt" file (no idea if it compiles without currently, but do it please just for our sanity) - at some point when we finally have separate DLLs it makes sense to compile without the driver when only minimal namespace is considered which doesn't require it

Reply:
why are you calling it "vulkdan driver" this is about the SDK, the whole point of the Vulkan Loader and ICD is that there's a separate vulkan loader lib that handles loading different drivers for different gpus.

hence you should be able to compile a vulkan app even if no Vulkan GPU or driver installed on the system

Reply:
indeed and I think I just blindly copy pasted your words https://github.com/Devsh-Graphics-Programming/Nabla/commit/7e00fd639581416363f5c970eff568b229f79fa2 but yes, its all about SDK so maybe lets call the file "vulkan-sdk-not-found.txt" or sth instead

Reply:
mea maxima culpa, yes I followed the Irrlicht tradition of misnaming things in back in 2020